week5/ciper/ciper.rb

```diff
@@ -0,0 +1,57 @@
+# converting string by caesar ciper
+  
+LOWER_ALPHABET = ('a'..'z').to_a
+UPPER_ALPHABET = ('A'..'Z').to_a
+ALPHABET_SIZE = 26
+
+class CaesarCiper
+  class IntegerError < StandardError; end
+
+  def initialize(ciper_string, ciper_number)
+    @ciper_string = ciper_string
+    @ciper_number = ciper_number
+  end
+
+  def call 
+    check_validate
+    converting_string(:+)
+  rescue IntegerError => e
+    e.message
+  end
+
+  def unencrypted_text
+    unencrypted_answer = converting_string(:-)
+  end
+
+  private
+
+  def converting_string(operator)
+    @answer = ""
+    @ciper_string.each_char do |char|
```

Comment:
please use `reduce` https://medium.com/@meganr.edmonds/the-multiple-uses-of-the-reduce-method-in-ruby-7b79382c000a